tests/unit/ops/transformer/inference/test_quant.py

```diff
@@ -0,0 +1,38 @@
+import torch
+import pytest
+from deepspeed.ops import op_builder
+
+quantizer_cuda_module = op_builder.QuantizerBuilder().load()
+
+
+def allclose(x, y):
+    assert x.dtype == y.dtype
+    rtol, atol = {torch.float32: (1e-2, 1e-3), torch.float16: (1e-2, 1e-3)}[x.dtype]
+    return torch.allclose(x, y, rtol=rtol, atol=atol)
+
+
+def quantize_dequantize_ref(inputs, bit, num_groups=1):
+    # quantize
+    q_range = 2**bit
+    input_flat = inputs.float().reshape(num_groups, -1).contiguous()
+    input_flat = torch.nan_to_num(input_flat, nan=0.0)
+    input_min = input_flat.amin(-1, keepdim=True)
+    input_max = input_flat.amax(-1, keepdim=True)
+
+    scale = q_range / (2 * torch.max(input_min.abs(), input_max.abs()))
+    input_flat = (input_flat * scale).round().clamp(-q_range // 2, q_range // 2 - 1)
+    # dequantize
+    return input_flat.reshape(inputs.shape).to(torch.int8) / scale.view(-1).to(
+        torch.float16)
+
+
+@pytest.mark.inference
+@pytest.mark.parametrize("input_tensor", torch.rand(8, 8, dtype=torch.float16).cuda())
```

Comment:
I think we could benefit from some more test cases here. Looking at the kernels, we should test that it partitions in the correct number of groups correctly and things like that as well as larger tensor/higher dimensioned Tensors.

Reply:
@cmikeh2 added more tests and also explicitly mentioned the group size boundary in comment.